src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -243,17 +251,35 @@ public function canFilterAnyColumns($gridField)
      * Generate a search context based on the model class of the of the GridField
      *
      * @param GridField $gridfield
-     * @return \SilverStripe\ORM\Search\SearchContext
+     * @return SearchContext
      */
     public function getSearchContext(GridField $gridField)
     {
         if (!$this->searchContext) {
-            $this->searchContext = singleton($gridField->getModelClass())->getDefaultSearchContext();
+            $modelClass = $gridField->getModelClass();
+            $singleton = singleton($modelClass);
+            if (!$singleton->hasMethod('getDefaultSearchContext')) {
+                throw new LogicException(
+                    'Cannot dynamically instantiate SearchContext. Pass the SearchContext to setSearchContext()'
+                    . " or implement a getDefaultSearchContext() method on $modelClass"
+                );
+            }
+            $this->searchContext = $singleton->getDefaultSearchContext();
```

Comment:
Doesn't have to respect `getThrowExceptionOnBadDataType` because this would have already thrown an exception when it tried to call the non-existent `getDefaultSearchContext()`